functions.js

```diff
@@ -0,0 +1,61 @@
+const mysql = require('promise-mysql')
+const {set} = require("express/lib/application");
+const connection = mysql.createConnection({
+    host: '127.0.0.1',
+    user: 'root',
+    password: 'password',
+    database: 'tube_info'
+})
+const getTubes = () => async (request, response) => {
+    const db = await connection
+    const tubeStops =  await db.query('SELECT `name`, `code`, `line`, `id` FROM `tube_info` ORDER BY `line`, `id`;')
+    response.json(tubeStops)
+}
+
+const getRoute = () => async (request, response) => {
+    const from = request.query.from
+    const to = request.query.to
+
+    const fromCode = from.slice(-3)
+    const toCode = to.slice(-3)
```

Comment:
I'm pretty sure you can avoid having to do this by _only_ sending the code from the front end